src/network/networkd-link.c

```diff
@@ -218,6 +233,26 @@ static IPv6PrivacyExtensions link_ipv6_privacy_extensions(Link *link) {
         return link->network->ipv6_privacy_extensions;
 }
 
+static int link_disable_ipv6(Link *link) {
+        const char *p = NULL;
+        int r;
+
+        /* Make this a NOP if IPv6 is not available */
+        if (!socket_ipv6_is_supported())
+                return 0;
+
+        if (link->flags & IFF_LOOPBACK)
+                return 0;
+
+        p = strjoina("/proc/sys/net/ipv6/conf/", link->ifname, "/disable_ipv6");
+
+        r = write_string_file(p, one_zero("1"), WRITE_STRING_FILE_VERIFY_ON_FAILURE);
+        if (r < 0)
+                log_link_warning_errno(link, r, "Cannot disale IPv6 for interface: %m");
+
```

Comment:
Hmm, I think we should also write "0" to this setting in case Ipv6 is not disabled, so that transitions from ipv6-mode to non-ipv6 can be handled nicely. i.e. maybe rename this call to "link_enable_ipv6()" and make it check link_ipv6_enabled() on its own, and write the inverted return value of that into the file, regardless if true or false...